### PR TITLE
Adding new rule SNS_TOPIC_EMAIL_SUB_IN_DOMAINS

### PR DESCRIPTION
**Pull request by vmunugal**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*
Issue #209 
*Description of changes:*

### CODE REVIEW

**Comment by 5u5hma** [comment]
@jongogogo Please review the code and let us know for any changes.

File: python/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS.py [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    sns_client = get_client('sns', event)
+    evaluations = []
+    topic = ''
+    subscription_endpoints = get_all_subscriptions(sns_client)
+    if not subscription_endpoints:
+        return None
+    for endpoint in subscription_endpoints:
+        domain_extract = subscription_endpoints[endpoint].split('@')
+        topic = endpoint+ ':' + subscription_endpoints[endpoint]
```
Comment (jongogogo) [inline python/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS.py]: missing a space between "endpoint" and "+"

**Comment by 5u5hma** [comment]
@jongogogo the requested changes have been made. Please review the same and lets us know for any further changes.

File: python/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS.py [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        for subscription in subscriptions_list['Subscriptions']:
+            if subscription['Protocol'] in valid_protocols:
+                dict_to_return[subscription['TopicArn']] = subscription['Endpoint']
+        if 'NextToken' not in subscriptions_list:
+            return dict_to_return
+        subscriptions_list = client.list_subscriptions(NextToken=subscriptions_list['NextToken'])
+
+def evaluate_parameters(rule_parameters):
+    if not rule_parameters['domainNames']:
+        raise ValueError('Atleast one domain name is required as input')
```
[the remark this answers is not in the document]
Comment (jongogogo) [reply python/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS/SNS_TOPIC_EMAIL_SUB_IN_DOMAINS.py]: @5u5hma @vmunugal Please address this comment

**Comment by jongogogo** [comment]
See in comments above

**Comment by vmunugal** [comment]
@jongogogo that comment has been addressed. Please let us know for any further changes.

**Pull request merged**